include/static-pages.php

```diff
@@ -122,23 +124,37 @@ public function page_link( $link, $id ) {
 	}
 
 	/**
-	 * Adds page_on_front and page_for_posts properties to the language objects.
+	 * Adds `page_on_front` and `page_for_posts` properties to language data before the object is created.
 	 *
-	 * @since 1.8
+	 * @since 3.4
 	 *
-	 * @param PLL_Language[] $languages The list of languages.
-	 * @param PLL_Model      $model     The instance of PLL_Model.
-	 * @return PLL_Language[]
+	 * @param array $additional_data Array of language additional data.
+	 * @param array $language        Language data.
+	 * @return array Language data with additional `page_on_front` and `page_for_posts` options added.
 	 */
-	public static function pll_languages_list( $languages, $model ) {
-		if ( 'page' === get_option( 'show_on_front' ) ) {
-			foreach ( $languages as $k => $language ) {
-				$languages[ $k ]->page_on_front = $model->post->get( get_option( 'page_on_front' ), $language );
-				$languages[ $k ]->page_for_posts = $model->post->get( get_option( 'page_for_posts' ), $language );
-			}
+	public function set_static_pages( $additional_data, $language ) {
+		$additional_data['page_on_front']  = $this->get_static_page_translation( 'page_on_front', $language );
+		$additional_data['page_for_posts'] = $this->get_static_page_translation( 'page_for_posts', $language );
+
+		return $additional_data;
+	}
+
+	/**
+	 * Returns id of the static page translation.
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $static_page  Static page option name; `page_on_front` or `page_for_posts`.
+	 * @param array  $language     Language data.
+	 * @return int Id of the static page translation.
+	 */
+	public function get_static_page_translation( $static_page, $language ) {
```

Comment:
I suggest to rename the method. It's useless to add `_static_page` in the method name when it's already in the class name.
```suggestion
	public function get_translation( $static_page, $language ) {
```

Reply:
Done in 84bcffcab9eb235930510ea1342fd4e9c76028b6